sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsClientLiveTestBase.cs

```diff
@@ -43,5 +51,23 @@ public TextAnalyticsClientLiveTestBase(bool isAsync, TextAnalyticsClientOptions.
                 return InstrumentClient(new TextAnalyticsClient(endpoint, credential, InstrumentClientOptions(options)));
             }
         }
+
+        // This has been added to stop the custom tests to run forever while we
+        // get more reliable information on which scenarios cause timeouts.
+        // Issue https://github.com/Azure/azure-sdk-for-net/issues/25152
+        internal async Task PollUntilTime(AnalyzeActionsOperation operation)
+        {
+            TimeSpan pollingInterval = TimeSpan.FromSeconds(10);
+            int timeout = (int)TimeSpan.FromMinutes(20).TotalMilliseconds;
+            using CancellationTokenSource cts = new CancellationTokenSource(timeout);
+            try
+            {
+                await operation.WaitForCompletionAsync(pollingInterval, cts.Token);
```

Comment:
I know we were talking about this the other day, trying to figure out a way to set a timeout for `WaitForCompletionAsync`. Completely forgot we had cancellation tokens!